### PR TITLE
Forknet Setup + Forknet morpho wsteth/usdc deploy + Forknet eeth deploy

### PR DESCRIPTION
**Pull request by mcclurejt**
# Resolved Issues

- Excessive time being spent on Sepolia mocks.
- Inaccurate testing environment.

# Description

- Adjusting scripting to enable deploying to remote fork.
- Copies existing deployments from deployments.json and writes new deployed contracts under deployments.local.json
- Expanding minting functions.
- Use same config for mainnet and testing network

# Review Checklists

Please check each item **before approving** the pull request. While going
through the checklist, it is recommended to leave comments on items that are
referenced in the checklist to make sure that they are reviewed. If there are
multiple reviewers, copy the checklists into sections titled `## [Reviewer Name]`.
If the PR doesn't touch Solidity, the corresponding checklist can
be removed.

## [[Reviewer Name]]

- [ ] **Tokens**
    - [ ] Do all `approve` calls use `forceApprove`?
    - [ ] Do all `transfer` calls use `safeTransfer`?
    - [ ] Do all `transferFrom` calls use `msg.sender` as the `from` address?
        - [ ] If not, is the function access restricted to prevent unauthorized
              token spend?
- [ ] **Low-level calls (`call`, `delegatecall`, `staticcall`, `transfer`, `send`)**
    - [ ] Is the returned `success` boolean checked to handle failed calls?
    - [ ] If using `delegatecall`, are there strict access controls on the
          addresses that can be called? It shouldn't be possible to `delegatecall`
          arbitrary addresses, so the list of possible targets should either be
          immutable or tightly controlled by an admin.
- [ ] **Reentrancy**
    - [ ] Are functions that make external calls or transfer ether marked as `nonReentrant`?
        - [ ] If not, is there documentation that explains why reentrancy is
              not a concern or how it's mitigated?
- [ ] **Gas Optimizations**
    - [ ] Is the logic as simple as possible?
    - [ ] Are the storage values that are used repeatedly cached in stack or
          memory variables?
    - [ ] If loops are used, are there guards in place to avoid out-of-gas
          issues?
- [ ] **Visibility**
    - [ ] Are all `payable` functions restricted to avoid stuck ether?
- [ ] **Math**
    - [ ] Is all of the arithmetic checked or guarded by if-statements that will
          catch underflows?
    - [ ] If `Safe` functions are altered, are potential underflows and
          overflows caught so that a failure flag can be thrown?
    - [ ] Are all of the rounding directions clearly documented?
- [ ] **Testing**
    - [ ] Are there new or updated unit or integration tests?
    - [ ] Do the tests cover the happy paths?
    - [ ] Do the tests cover the unhappy paths?
    - [ ] Are there an adequate number of fuzz tests to ensure that we are
          covering the full input space?

### CODE REVIEW

File: deployments.json [hunk 1]
```diff
@@ -836,6 +836,37 @@
             "contract": "MorphoBlueHyperdrive",
             "address": "0xd41225855A5c5Ba1C672CcF4d72D1822a5686d30",
             "timestamp": "2024-08-08T15:43:26.033Z"
+        },
```
Comment (mcclurejt) [inline deployments.json:839]: just fyi this is here because I forgot to commit/push it after deploying last week

File: .dockerignore [hunk 2]
```diff
@@ -6,5 +6,4 @@ node_modules/
 out/
 target/
 *.env
-deployments.json
```
Comment (mcclurejt) [inline .dockerignore:9]: need this in the image to bootstrap the mainnet_fork section

File: tasks/deploy/config/mainnet/eeth-182day.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    name: MAINNET_EETH_182DAY_NAME,
+    prefix: "EETH",
+    coordinatorAddress: async (hre) =>
+        hre.hyperdriveDeploy.deployments.byName(MAINNET_EETH_COORDINATOR_NAME)
+            .address,
+    deploymentId: keccak256(toBytes(MAINNET_EETH_182DAY_NAME)),
+    salt: toBytes32("0xababe"),
+    extraData: "0x",
+    contribution: CONTRIBUTION,
+    fixedAPR: parseEther("0.09"),
```
Comment (jalextowle) [inline tasks/deploy/config/mainnet/eeth-182day.ts:32]: This is 6.5% on Pendle.

File: tasks/deploy/config/mainnet/eeth-182day.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                CONTRIBUTION,
+            ]),
+        ]);
+        await pc.waitForTransactionReceipt({ hash: tx });
+    },
+    poolDeployConfig: async (hre) => {
+        return {
+            baseToken: ETH_ADDRESS,
+            vaultSharesToken: EETH_ADDRESS_MAINNET,
+            circuitBreakerDelta: parseEther("0.1"),
```
Comment (jalextowle) [inline tasks/deploy/config/mainnet/eeth-182day.ts:66]: We need to change this to 7.5%

File: tasks/deploy/config/mainnet/morpho-blue-wsteth-usdc-182day.ts [hunk 5]
```diff
@@ -66,8 +73,8 @@ export const MAINNET_MORPHO_BLUE_WSTETH_USDC_182DAY: HyperdriveInstanceConfig<"M
             hre.hyperdriveDeploy.deployments.byName(
                 MAINNET_MORPHO_BLUE_COORDINATOR_NAME,
             ).address,
-        deploymentId: toBytes32(
-            MAINNET_MORPHO_BLUE_WSTETH_USDC_182DAY_NAME.slice(0, 32),
+        deploymentId: keccak256(
+            toHex(MAINNET_MORPHO_BLUE_WSTETH_USDC_182DAY_NAME),
         ),
         salt: toBytes32("0x42080085"),
         extraData: morphoBlueParameters,
```
Comment (jalextowle) [inline tasks/deploy/config/mainnet/morpho-blue-wsteth-usdc-182day.ts:80]: Let's change the fixed rate to 3.5%

File: tasks/deploy/lib/constants.ts [hunk 6]
```diff
@@ -3,51 +3,88 @@ import { Address } from "viem";
...
     "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE" as Address;
 
-export const MAINNET_DAI_ADDRESS =
-    "0x6b175474e89094c44da98b954eedeac495271d0f" as Address;
+// ╭─────────────────────────────────────────────────────────╮
+// │ Mainnet Addresses                                       │
+// ╰─────────────────────────────────────────────────────────╯
 
-export const MAINNET_SDAI_ADDRESS =
-    "0x83F20F44975D03b1b09e64809B757c47f942BEeA" as Address;
```
Comment (jalextowle) [inline tasks/deploy/lib/constants.ts:10]: Did we get rid of this?

Comment (jalextowle) [reply tasks/deploy/lib/constants.ts:10]: Don't we need this for the sDAI integraiton?

Comment (jalextowle) [reply tasks/deploy/lib/constants.ts:10]: Nvm, I see it.

File: tasks/fork/maintain-rate.ts [hunk 7]
```diff
@@ -85,31 +84,29 @@ HyperdriveDeployBaseTask(
...
+            let txData = encodeFunctionData({
+                abi: (await hre.artifacts.readArtifact("ILiquidityPool")).abi,
+                functionName: "rebase",
+                args: [etherToAdd],
+            });
+            await tc.sendUnsignedTransaction({
+                from: EETH_MEMBERSHIP_MANAGER_ADDRESS_MAINNET,
+                to: EETH_LIQUIDITY_POOL_ADDRESS_MAINNET,
+                data: txData,
             });
```
Comment (jalextowle) [inline tasks/fork/maintain-rate.ts:105]: Is this what you had to do to avoid separate configs?

File: tasks/fork/maintain-rate.ts [hunk 8]
```diff
@@ -85,31 +84,29 @@ HyperdriveDeployBaseTask(
             // value is not recalculated.
 
             // For EETH, we can call rebase on the Etherfi contracts using an
```
Comment (jalextowle) [inline tasks/fork/maintain-rate.ts:86]: This comment looks wrong now.

File: tasks/fork/maintain-rate.ts [hunk 9]
```diff
@@ -85,31 +84,29 @@ HyperdriveDeployBaseTask(
...
-            // impersonated account.
-            const config = hre.network.config as HttpNetworkUserConfig;
-            config.accounts = "remote";
-            await tc.impersonateAccount({
-                address: MAINNET_ETHERFI_MEMBERSHIP_MANAGER,
-            });
```
Comment (jalextowle) [inline tasks/fork/maintain-rate.ts:93]: Why did we get rid of this?

**Review by jalextowle — APPROVED** [review]
Nice work dude! LGTM once the comments are addressed

File: tasks/fork/maintain-rate.ts (hunk 7 above)
Comment (mcclurejt) [reply tasks/fork/maintain-rate.ts:105]: Yup! Appears to have worked

File: tasks/fork/maintain-rate.ts (hunk 9 above)
Comment (mcclurejt) [reply tasks/fork/maintain-rate.ts:93]: not needed and I believe it was causing the multiple config issue.

**Pull request merged**